addons/website_sale/views/templates.xml

```diff
@@ -760,7 +760,7 @@
     </template>
 
     <template id="product_quantity" inherit_id="website_sale.product" customize_show="True" name="Select Quantity">
-      <xpath expr="//div[@id='add_to_cart_wrap']" position="before">
+      <xpath expr="//a[@id='add_to_cart']" position="before">
```

Comment:
IMO changing the structure should be avoided:

1) it will be different than 16 where it works so this fix will not be forward-ported
2) it will only be available to people that install 15.0 for the first time, or update website_sale
3) also since will only be available on module update, the JS will need to work for both case (if the XML is updated or not)

checking in 16.0 there is the new id `o_wsale_cta_wrapper` around the elements and we hide it (so it's a little similar to your solution) but I think iyou could just not change the XML and do when it's out of stock:

```js
$parent.find('#add_to_cart_wrap').prev('.css_quantity').addClass('d-none');
```
and at the same place as the `.removeClass('out_of_stock');` do:

```js
$parent.find('#add_to_cart_wrap').prev('.css_quantity').removeClass('d-none');
```